lxd/auth/driver_tls.go

```diff
@@ -108,12 +112,15 @@ func (t *tls) GetPermissionChecker(ctx context.Context, r *http.Request, entitle
 		return allowFunc(true), nil
 	}
 
-	certType, isNotRestricted, projectNames, err := t.certificateDetails(details.username())
+	username := details.username()
+	id, err := t.identities.Get(api.AuthenticationMethodTLS, details.username())
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Certificate %q missing from identity cache", username)
```

Comment:
Same point as above, we should be wrapping errors if we aren't checking them for specific meaning.